### PR TITLE
groups: make bloc actually "sudo"

### LINKED ISSUE #1502: notebook: deleting an old note in Tlon Hosting Support fails with sects error
I'm an admin and can't delete it. Nick wrote it and can't delete it either.

```
TypeError: Cannot read properties of undefined (reading 'sects')
    at vvt (https://zurbit.io/apps/groups/assets/index.6b96209d.js:1138:68854)
    at qH (https://zurbit.io/apps/groups/assets/index.6b96209d.js:93:53297)
    at Yie (https://zurbit.io/apps/groups/assets/index.6b96209d.js:97:8857)
    at Rie (https://zurbit.io/apps/groups/assets/index.6b96209d.js:97:964)
    at P6e (https://zurbit.io/apps/groups/assets/index.6b96209d.js:97:887)
    at _w (https://zurbit.io/apps/groups/assets/index.6b96209d.js:97:737)
    at UG (https://zurbit.io/apps/groups/assets/index.6b96209d.js:95:10947)
    at Kd (https://zurbit.io/apps/groups/assets/index.6b96209d.js:95:8466)
    at l8 (https://zurbit.io/apps/groups/assets/index.6b96209d.js:97:15965)
    at https://zurbit.io/apps/groups/assets/index.6b96209d.js:97:17629
```
Screenshot 2022-12-14 at 3 51 56 PM

### LINKED ISSUE #1556: key channels in UF Public group are unjoinable
Several channels in the UF Public group cannot be joined by new ships with the latest Groups OTA (`iq85j`). 

**To Reproduce**
1. Go to urbit.org/trial and launch a fresh moon. 
2. Open the web terminal and verify that it is in fact on Groups `iq85j`. Leave it open.
3. Go back to Landscape and open Groups, search for `~halbex-palheb`, join the UF group. 
4. Note the UI jank in the joining process, which this issue is not concerned with.
5. Go to All Channels in the newly joined group. Join Developer Reports, Ecosystem, Infrastructure — note how the UI says they're joined when they're obviously not.
6. Go back to the web terminal and note the error messages about those channels.

**Additional context:**
Stack trace for failed channel join:
```
%diary: on-agent on wire /diary/~halbex-palheb/infrastructure-6618/updates, %watch-ack
Failed subscription
watch-ack
/app/diary/hoon:<[815 5].[822 12]>
/app/diary/hoon:<[814 5].[822 12]>
/app/diary/hoon:<[635 23].[635 38]>
/app/diary/hoon:<[634 5].[639 7]>
/app/diary/hoon:<[633 5].[639 7]>
/app/diary/hoon:<[368 13].[368 66]>
/app/diary/hoon:<[368 5].[368 66]>
/app/diary/hoon:<[367 5].[368 66]>
/app/diary/hoon:<[359 3].[375 5]>
/app/diary/hoon:<[358 3].[375 5]>
/app/diary/hoon:<[57 12].[57 28]>
/app/diary/hoon:<[57 7].[57 28]>
/app/diary/hoon:<[56 5].[58 17]>
/app/diary/hoon:<[55 5].[58 17]>
/sys/vane/gall/hoon:<[1.227 9].[1.227 37]>
%diary: on-poke with mark %flag
```

Video of the entire process:

https://user-images.githubusercontent.com/550978/208174024-28d9fc47-d16e-404d-862c-d975151b003d.mov

### PR DESCRIPTION
**Pull request by arthyn**
This makes sure that sects contained in bloc actually act as superusers allowing them full permissions specifically around channel reading/writing. Also adds a poke to reset all channel permissions

Should fix #1502 and gives avenue to fix #1556

### CODE REVIEW

**Comment by jamesacklin** [comment]
Please test #1552 as well.

**Comment by arthyn** [comment]
Added fix for #1552

File: ui/src/groups/ChannelIndex/ChannelIndex.tsx [hunk 1]
```diff
@@ -135,7 +137,10 @@ function GroupChannel({
     navigate(channelHref(groupFlag, nest));
   }, [groupFlag, joined, navigate, nest]);
 
-  if (!group || (channel.readers.includes('admin') && !isAdmin)) {
+  const canRead = canReadChannel(channel, vessel, group?.bloc);
+  console.log(canRead, channel, vessel, group?.bloc);
```
Comment (tomholford) [inline ui/src/groups/ChannelIndex/ChannelIndex.tsx:141]: Still needed?

File: ui/src/groups/ChannelIndex/ChannelIndex.tsx [hunk 2]
```diff
@@ -304,6 +309,8 @@ export default function ChannelIndex({ title }: ViewProps) {
   const isMobile = useIsMobile();
   const BackButton = isMobile ? Link : 'div';
 
+  console.log(filteredSections, sectionedChannels);
```
Comment (tomholford) [inline ui/src/groups/ChannelIndex/ChannelIndex.tsx:312]: Still needed?

File: ui/src/groups/ChannelIndex/ChannelIndex.tsx (hunk 1 above)
Comment (arthyn) [inline ui/src/groups/ChannelIndex/ChannelIndex.tsx:141]:
```suggestion
```

**Review by tomholford — APPROVED** [review]
LGTM. Is the `reset-all-perms` poke consumed anywhere?

**Comment by arthyn** [comment]
@tomholford it's meant to be used manually, same with reset-group-perms. We converged on manual resolution of channel permissions instead of automatic.

File: desk/app/groups.hoon [hunk 3]
```diff
@@ -1245,7 +1313,7 @@
     ::
         %del
       ?<  &((~(has in ships) our.bowl) =(p.flag our.bowl))
-      ?>  ?|(=(p.flag src.bowl) (~(has in ships) src.bowl))
+      ?>  ?|(go-is-bloc =(p.flag src.bowl) (~(has in ships) src.bowl))
```
Comment (ryjm) [inline desk/app/groups.hoon:1316]: (nitpicky feel free to ignore) would prefer tall form here, slightly easier to parse

**Review by ryjm — APPROVED** [review]
just some small style suggestions otherwise LGTM

**Comment by jamesacklin** [comment]
Did this resolve #1556?

**Pull request merged**